src/core/expression/qgsexpressionfunction.cpp

```diff
@@ -6730,6 +6750,126 @@ static QVariant executeGeomOverlay( const QVariantList &values, const QgsExpress
 
     if ( ! relationFunction || ( geometry.*relationFunction )( feat2.geometry() ) ) // Calls the method provided as template argument for the function (e.g. QgsGeometry::intersects)
     {
+
+      if ( isIntersectsFunc && ( minOverlap != -1 || minInscribedCircleRadius != -1 ) )
+      {
+        const QgsGeometry intersection { geometry.intersection( feat2.geometry() ) };
+
+        switch ( intersection.type() )
+        {
+          case QgsWkbTypes::GeometryType::PolygonGeometry:
+          {
+            // Check min overlap for intersection (if set)
+            if ( intersection.isMultipart() )
+            {
+              bool testResult { false };
+              for ( auto it = intersection.const_parts_begin(); ! testResult && it != intersection.const_parts_end(); ++it )
+              {
+                const QgsPolygon *geom = qgsgeometry_cast< const QgsPolygon * >( *it );
+                // qDebug() << "Area" << feat2.id() << geom->area();
+                if ( minOverlap != -1 )
+                {
+                  if ( geom->area() >= minOverlap )
+                  {
+                    testResult = true;
```

Comment:
```suggestion
                    testResult = true;
                    break; 
```

we can shortcut and skip the other checks. Or are the checks supposed to be "AND" conditions instead of "OR"? (Either way, better document what happens when both are specified!)

Reply:
Yes, the checks must be ANDed.

Reply:
Can you add this clarification to the function documentation?

Reply:
I'm changing it to:

```json
{
      "arg": "min_overlap",
      "description": "defines an optional exclusion filter: for polygons a minimum area in current feature squared units for the intersection (if the intersection results in multiple polygons the intersection will be returned if at least one of the polygons has an area greater or equal to the value), for lines a minimum length in current feature units (if the intersection results in multiple lines the intersection will be returned if at least one of the lines has a length greater or equal to the value).",
      "optional": true
    },
    {
      "arg": "min_inscribed_circle_radius",
      "description": "defines an optional exclusion filter (for polygons only): minimum radius in current feature units for the maximum inscribed circle of the intersection (if the intersection results in multiple polygons the intersection will be returned if at least one of the polygons has a radius for the maximum inscribed circle greater or equal to the value).<br>Read more on the underlying GEOS predicate, as described in PostGIS <a href='https://postgis.net/docs/ST_MaximumInscribedCircle.html'>ST_MaximumInscribedCircle</a> function.<br>This argument requires GEOS >= 3.9.",
      "optional": true
    }
```

to me it's obvious that they are ANDed but if you have any idea about how the explicitly say it please let me know.